mesonbuild/scripts/clangformat.py

```diff
@@ -37,7 +37,10 @@ def parse_pattern_file(fname: Path) -> T.List[str]:
 
 def run_clang_format(exelist: T.List[str], fname: Path, check: bool) -> subprocess.CompletedProcess:
     before = fname.stat().st_mtime
-    ret = subprocess.run(exelist + ['-style=file', '-i', str(fname)])
+    args = ['-style=file', str(fname)]
+    if check:
+        args = ['-i'] + args
+    ret = subprocess.run(exelist + args)
```

Comment:
You inverted the logic here, should be `if not check`.

It's a bit more complicated than that. When not using `-i` it will print the formatted file on stdout, you have to compare that output with the original file to know if it was modified because the mtime check below won't work since it did not modify the file.

In retrospect relying on mtime was maybe a cheap trick, we should probably never use `-i` and always do the diff between original file and stdout. Then `if not check and there_is_a_diff: rewrite_the_file(stdout)`.